client.go

```diff
@@ -202,7 +199,6 @@ func New(options ...Option) *Client {
 		dnsProvider:       &noopDNSProvider{output: os.Stdout},
 		progressListener:  &NoopProgressListener{},
 		pipelinesProvider: &noopPipelinesProvider{},
-		repositoriesPath:  filepath.Join(config.Path(), "repositories"),
```

Comment:
Repository path is now entirely optional from the standpoint of the core library.  When not provided, it simply presumes all repository-related operations are with builtin repositories only, with the exception of things like "add", which will of course error with a message.  This allevaites the problem where, in some circumstances, the client would start creating paths on disk when not necessary.